lib/beaker/dsl/install_utils/pe_utils.rb

```diff
@@ -361,9 +361,12 @@ def do_install hosts, opts = {}
             pre30master = version_is_less(opts[:pe_ver] || master['pe_ver'], '3.0')
           end
 
+          agent_only_check_needed, hosts_agent_only, hosts_not_agent_only = \
+            create_hosts_agent_only_specified_arrays_helper(hosts, opts['pe_ver'])
```

Comment:
I don't doubt the setting of the default value, I just find the logic of this confusing.  You pass in the global, default it to 3.8 and then compare it to 3.99 - and then do a check on individual hosts pe versions.  Maybe this could be streamlined by just generating an array of all possible pe_ver (global and host) and checking to see if the largest is > 3.99?

pe_versions =( [] << opts['pe_ver'] << hosts.map{ |host| host['pe_ver']}).compact.flatten

Otherwise the steps are spread out in a weird way that makes the code hard to follow.